main.go

```diff
@@ -6,18 +6,24 @@ import (
 )
 
 func main() {
-	router := gin.Default()
-	var corsConfig = cors.DefaultConfig()
-	// TODO: for the time being, we allow requests from all origins.
+    router := gin.Default()
+    var corsConfig = cors.DefaultConfig()
+    // TODO: for the time being, we allow requests from all origins.
 	// We can consider narrowing this if/when the Project Ricotta front-end has
 	// specific known origin domains that we want to restrict requests to originating from.
-	corsConfig.AllowAllOrigins = true
-	corsConfig.AllowHeaders = append(corsConfig.AllowHeaders, "authorization")
-	router.Use(cors.New(corsConfig))
-	router.POST("/login", postUserAuthorization)
-	router.GET("/profile", getCurrentUserProfile)
-	router.GET("/profile/:userID", getUserProfileByID)
-	router.POST("/profile", postUserProfile)
+    corsConfig.AllowAllOrigins = true
+    corsConfig.AllowHeaders = append(corsConfig.AllowHeaders, "authorization")
+    router.Use(cors.New(corsConfig))
+    
+    router.POST("/login", postUserAuthorization)
+    router.GET("/profile", getCurrentUserProfile)
+    router.GET("/profile/:userID", getUserProfileByID)
+    router.POST("/profile", postUserProfile)
+    router.POST("/register", postUserRegistration)
```

Comment:
I'd really suggest hooking the new user registration up to the existing POST on /profile, rather than making a new and duplicate /register API.
Also, best practice to have REST API endpoints be nouns whenever possible, rather than verbs. (Yes, "login" is ambiguous, I know). It's customary and recommended to think in terms of nouns and then the HTTP verbs (GET, POST, PATCH, etc.) as the actual action to perform.
documentation\bechamel-api.yaml has the most current proposal for the API. I believe if you hook what you wrote in api_register.go (and move that to api_profile.go, see separate comment) then it makes for a more consistent API both in terms of what's proposed and in terms of recommended practices.